source/frontend/src/features/properties/map/shared/SidebarFooter.tsx

```diff
@@ -18,18 +20,31 @@ const SidebarFooter: React.FunctionComponent<ISidebarFooterProps> = ({
   onSave,
   onCancel,
   isOkDisabled,
+  isMissingFields,
 }) => {
+  const [errorMessage, setErrorMessage] = useState('');
```

Comment:
Whenever implementing changes, we always want to implement the minimum possible added state, as every usage of state complicates the application and generally requires testing.

Additionally, we should avoid using state in non-container components.
files with *container* in the name should be used to hold state and contain state manipulation functions. 
all other components are considered "display" components and should not contain state. If view components require the ability to manipulate state they should do that via props.

Reply:
Makes sense! However, is useEffect okay to use in a "display" component?

Reply:
In general, no. a useEffect should live close to the state that it manipulates.